vault-client/init.sls

```diff
@@ -0,0 +1,26 @@
+{%- if 'vault' in pillar %}
+
+{%- set vault = salt['pillar.get']('vault:lookup', {}) %}
+{%- set os_family = salt['grains.get']('os_family') %}
+
+# Include the module to install the dependencies base on the OS type. 
+{%- if os_family == 'RedHat' %}
```

Comment:
Looks like we don't need the lx/win split anymore, with that change to the pip install. I'd like to split this into three sls files...

vault-client/install.sls - installs the packages needed (`hvac`)
vault-client/authenticate.sls - calls the `vault-client.authenticated` state
vault-client/init.sls - has an include block that calls both `.install` and `.authenticate`